### PR TITLE
Add three band equalizer

### PR DESCRIPTION
**Pull request by haileys**

### CODE REVIEW

File: src/module/eq_three.rs [hunk 1]
```diff
@@ -0,0 +1,170 @@
+use std::f64;
+
+use mixlab_protocol::EqThreeParams;
+
+use crate::engine::{Sample, SAMPLE_RATE, ZERO_BUFFER_MONO};
+use crate::module::{ModuleT, LineType, Terminal};
+
+const FREQ_LO: f64 = 420.0;
+const FREQ_HI: f64 = 2700.0;
```
Comment (haileys) [inline src/module/eq_three.rs:9]: Hardcoded bands for now. It would be good to parameterise them

File: src/module/eq_three.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    pub fn set_freq(&mut self, freq: f64) {
+        self.freq = 2.0 * f64::sin(f64::consts::PI * freq / (SAMPLE_RATE as f64));
+    }
+
+    pub fn pump(&mut self, sample: f64) -> f64 {
+        self.poles[0] += self.freq * (sample - self.poles[0]) + VSA;
+        self.poles[1] += self.freq * (self.poles[0] - self.poles[1]);
+        self.poles[2] += self.freq * (self.poles[1] - self.poles[2]);
+        self.poles[3] += self.freq * (self.poles[2] - self.poles[3]);
```
Comment (haileys) [inline src/module/eq_three.rs:120]: I have got absolutely no clue how this works but it does the job

**Comment by bjeanes** [comment]
I wish I understood this. Like... at all.

File: src/module/eq_three.rs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        let mut bytes = Vec::new();
+
+        for float in floats {
+            bytes.extend(&float.to_le_bytes());
+        }
+
+        bytes
+    }
+
+    #[test]
```
Comment (bjeanes) [inline src/module/eq_three.rs:149]: This reminds me.... I wanted to add a test for the PWM stuff I am adding in #34

**Pull request merged**